server/services/draftsService.test.ts

```diff
@@ -0,0 +1,174 @@
+import { Request } from 'express'
+import { mocked } from 'ts-jest/utils'
+import uuid from 'uuid'
+import DraftsService from './draftsService'
+
+jest.mock('uuid')
+const mockedUuid = mocked(uuid)
+
+describe(DraftsService, () => {
+  function createRequestWithSession(session: unknown): Request {
+    return { session } as unknown as Request
+  }
+
+  const clock = { now: () => new Date('2021-04-01T10:25:00Z') }
+
+  describe('.createDraft', () => {
+    it('generates a random UUID and stores the data in the session under a key derived from that UUID', () => {
+      mockedUuid.v4.mockReturnValue('2dc7ef56-58dd-4339-9924-c33318738068')
+      const req = createRequestWithSession({})
+
+      const draftsService = new DraftsService(clock)
+
+      const draft = draftsService.createDraft('someExampleType', { a: 1, b: 2 }, req)
+
+      expect(draft).toEqual({
+        id: '2dc7ef56-58dd-4339-9924-c33318738068',
+        type: 'someExampleType',
+        createdAt: new Date('2021-04-01T10:25:00Z'),
+        updatedAt: new Date('2021-04-01T10:25:00Z'),
+        data: { a: 1, b: 2 },
+      })
+
+      expect(req.session['draft-2dc7ef56-58dd-4339-9924-c33318738068']).toEqual({
```

Comment:
nit: i think it would be clearer to use a seperator that's not in the uuid here e.g. `draft:...` or `draft/...`

Reply:
Sure thing, I'll change it.